src/AracariNode.ts

```diff
@@ -0,0 +1,153 @@
+import { Instruction, InstructionType, MinimalNode, NodeType } from "./types";
+import { Aracari } from "./Aracari";
+
+interface AracariNodeOptions {
+  originalNode: MinimalNode;
+  parentNode?: AracariNode;
+  onUpdate?: (instruction: Instruction) => void;
+  root?: Aracari;
+  id?: string;
+}
+
+const noop = () => {};
+
+export class AracariNode {
+  __type = "ARACARI_NODE";
+  nodeType: number;
+  _content?: string;
+  id?: string;
+  childNodes: AracariNode[];
+  parentNode?: AracariNode;
+  originalNode: MinimalNode;
+  onUpdate: (instruction: Instruction) => void;
+  root?: Aracari;
+
+  // Need someway to load this in the background process.
+  constructor({
+    originalNode,
+    parentNode,
+    onUpdate,
+    root,
+    id,
+  }: AracariNodeOptions) {
+    this.nodeType = originalNode.nodeType;
+    this.root = root;
+    this.id = id;
+    this.childNodes = originalNode.childNodes
+      ? Array.from(originalNode.childNodes).map((node) =>
+          AracariNode.from({
+            originalNode: node,
+            parentNode: this,
+            onUpdate,
+            root,
+          })
+        )
+      : [];
+    this.parentNode = parentNode;
+    if (this.nodeType === NodeType.Text) {
+      this.textContent = originalNode.textContent;
+    }
+
+    this.onUpdate = onUpdate ?? noop;
+  }
+
+  get textContent() {
+    return this._content;
+  }
+
+  set textContent(text) {
+    if (this.nodeType === NodeType.HTMLElement) {
+      const node = AracariNode.from({
+        originalNode: {
+          nodeType: NodeType.Text,
+          textContent: text,
+          childNodes: null,
+        },
+        parentNode: this,
+      });
+      this.onUpdate({
+        target: this.id ?? null,
+        type: InstructionType.CreateText,
+        value: node,
+      });
+      // Need update to be able to append this to this element
```

Comment:
Remove this comment, was able to add the id of the node as the target to supplement this.